agent/task.go

```diff
@@ -121,12 +122,17 @@ func (a *Agent) checkIn(ctx context.Context, tc *taskContext, command command.Co
 		return
 	}
 
-	select {
-	case resetIdleTimeout <- duration:
-		tc.currentCommand = command
-		tc.logger.Execution().Infof("Command timeout set to %v", duration.String())
-	default:
-	}
+	resetIdleTimeout <- duration
+	tc.logger.Execution().Infof("Command timeout set to %s", duration.String())
+}
+
+func (tc *taskContext) setCurrentCommand(command command.Command) {
+	tc.currentCommand = command
+	tc.logger.Execution().Infof("Current command set to '%s'", tc.currentCommand.DisplayName())
```

Comment:
we should probably also log task type here?

Reply:
Done.